### PR TITLE
Added AWS_S3_HOST environment variable workaround to user_guide for i…

### LINKED ISSUE #26195: AWS_S3_HOST environment variable no longer available
In the [What's New - Other enhancements](https://pandas.pydata.org/pandas-docs/version/0.21/whatsnew.html#whatsnew-0180-enhancements-other) section from the 18.0 release it says that you can define the environment variable `AWS_S3_HOST`. 

Therefore, when setting this environment variable, together with `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` I expected the following code to work.

```python
import pandas as pd
df = pd.DataFrame(dict(a=[1, 2, 3], b=[4, 5, 6]))
df.to_csv(fs.open('s3://bucket/key', mode='w'))
```

The script crashes with:

```
Exception ignored in: <bound method S3File.__del__ of <S3File bucket/key>>
Traceback (most recent call last):
  File "/home/xxx/.local/share/virtualenvs/xxx/lib/python3.6/site-packages/s3fs/core.py", line 1518, in __del__
    self.close()
  File "/home/xxx/.local/share/virtualenvs/xxx/lib/python3.6/site-packages/s3fs/core.py", line 1496, in close
    raise_from(IOError('Write failed: %s' % self.path), e)
  File "<string>", line 3, in raise_from
OSError: Write failed: bucket/key
```
This is the exact same error as when I don't specify the variable `AWS_S3_HOST` at all.

I managed to get a working solution by directly interacting directly with `s3fs`:
```python
import os
import pandas as pd
import s3fs

fs = s3fs.S3FileSystem(client_kwargs={'endpoint_url': os.environ["AWS_S3_HOST"]})
df = pd.DataFrame(dict(a=[1, 2, 3], b=[4, 5, 6]))
df.to_csv(fs.open('s3://bucket/key', mode='w'))
```

Looks like this feature got (accidentally?) removed in `https://github.com/pandas-dev/pandas/commit/dc4b0708f36b971f71890bfdf830d9a5dc019c7b#diff-a37b395bed03f0404dec864a4529c97dL94`, when they switched from `boto` to `s3fs`.
Is it planned to support this environment variable again?
I'm sure that many companies, e.g. like the one I work for, hosts their own s3 server, e.g. with Minio, and don't use Amazon.

### PR DESCRIPTION
**Pull request by kvn4**
…ssue #26195

- [ ] closes #26195(Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/user_guide/io.rst [hunk 1]
```diff
@@ -2986,7 +3003,7 @@ Read a URL with no options:
 Read in the content of the "books.xml" file and pass it to ``read_xml``
 as a string:
 
-.. ipython:: python
+.. ipython:: python :okexcept:
```
Comment (mroeschke) [inline doc/source/user_guide/io.rst:3006]: Why was this added?

File: doc/source/user_guide/io.rst [hunk 2]
```diff
@@ -1712,9 +1712,26 @@ at `fsimpl1`_ for implementations built into ``fsspec`` and `fsimpl2`_
...
+
+.. code-block:: python
+
+   def s3so(worker_id):
+      worker_id = "5" if worker_id == "master" else worker_id.lstrip("gw")
+      return dict(client_kwargs={"endpoint_url": f"http://127.0.0.1:555{worker_id}/"})
+
+   roundtripped_df = pd.read_json(
+         "s3://pandas-test/test-1", compression=compression, storage_options=s3so,
+      )
```
Comment (mroeschke) [inline doc/source/user_guide/io.rst:1728]:
```suggestion
   storage_options = {"client_kwargs": {"endpoint_url": "http://127.0.0.1:5555"}}}
   df = pd.read_json("s3://pandas-test/test-1", storage_options=storage_options)
```

File: doc/source/user_guide/io.rst [hunk 3]
```diff
@@ -1712,9 +1712,26 @@ at `fsimpl1`_ for implementations built into ``fsspec`` and `fsimpl2`_
 for those not included in the main ``fsspec``
 distribution.
 
-You can also pass parameters directly to the backend driver. For example,
-if you do *not* have S3 credentials, you can still access public data by
-specifying an anonymous connection, such as
+You can also pass parameters directly to the backend driver. Because we can no
+longer utilize the ``AWS_S3_HOST`` environment variable, we can directly define a
```
Comment (mroeschke) [inline doc/source/user_guide/io.rst:1716]:
```suggestion
You can also pass parameters directly to the backend driver. Since ``fsspec`` does not
utilize the ``AWS_S3_HOST`` environment variable, we can directly define a
```

File: doc/source/user_guide/io.rst (hunk 1 above)
Comment (kvn4) [reply doc/source/user_guide/io.rst:3006]: Hi Matthew, thanks for your help. I added this initially because I was having errors when making, but I will remove this and repush

**Comment by kvn4** [comment]
Thank you for the suggestions @mroeschke, I updated and pushed to reflect the suggestions.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @kvn4
